co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureMessageAmbassador.java

```diff
@@ -149,44 +138,120 @@ private synchronized void receiveInteraction(InfrastructureV2xMessageReception i
         log.info(infrastructureV2xMessageReception.getReceiverID() + " received V2X messages at time: "
                 + infrastructureV2xMessageReception.getTime() + ".");
         log.info("The received message is " + infrastructureV2xMessageReception.getMessage() + " .");
+    }
 
+    /**
+     * Creates an Ad-Hoc configuration object to represent the configuration of the
+     * Ad-Hoc interface used for communication between the infrastructure instance
+     * and other vehicles or components, and sends it to the RTI for exchange.
+     *
+     * @param reg the infrastructure registration message received from the RTI
+     */
+    private final void onDsrcRegistrationRequest(InfrastructureRegistrationMessage reg) {
```

Comment:
Declaring this "final" is probably unnecessary, since we're unlikely to subclass this. Honestly probably better to explicitly declare things final more often, but it seems a bit inconsistent with the rest of what's currently written.

Also, it's super super minor, but something about passing the whole InfrastructureRegistrationMessage as the input here seems a bit off to me. I think slightly better would be either passing just the infrastructure ID (since that's all that's needed, data-wise) or passing in the actual InfrastructureInstance or something. As it is now, you could call this method prior to the `onRsuRegistration...` method below which I think would result in NS-3 not properly processing the registration. You could lean a bit more on the type-system here (by having the parameter here be of a type created by the `onRsuRegistration...` method to more clearly express the ordering just through the data types. Having them be the same input type in this case almost implies they work independently rather than sequentially. 

I don't think you need to make any changes here, but just wanted to share something to consider for future stuff.

Reply:
Might be worth adding to the doc comment that this should be called **after** the `onRsuRegistration...` method.